boilerplate/app/components/Button.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { ComponentType } from "react"
+import {
+  Pressable,
+  PressableProps,
+  PressableStateCallbackType,
+  StyleProp,
+  TextStyle,
+  ViewStyle,
+} from "react-native"
+import { TxKeyPath } from "../i18n"
+import { colors, typography } from "../theme"
+import { Text } from "./Text"
+
+type Presets = keyof typeof $viewPresets
+
+export interface ButtonProps extends PressableProps {
+  /**
+   * Text which is looked up via i18n.
+   */
+  tx?: TxKeyPath
+  /**
+   * The text to display if not using `tx` or nested components.
+   */
+  text?: string
+  /**
+   * An optional style override useful for padding & margin.
+   */
+  style?: StyleProp<ViewStyle>
+  /**
+   * An optional style override for the "pressed" state.
+   */
+  pressedStyle?: StyleProp<ViewStyle>
+  /**
+   * An optional style override for the button text.
+   */
+  textStyle?: StyleProp<TextStyle>
+  /**
+   * An optional style override for the button text when in the "pressed" state.
+   */
+  pressedTextStyle?: StyleProp<TextStyle>
+  /**
+   * One of the different types of text presets.
+   */
+  preset?: Presets
+  /**
+   * An optional component to render on the right side of the text.
+   * Example: `RightAccessory={(props) => <View {...props} />}`
+   */
+  RightAccessory?: ComponentType<{
+    style: StyleProp<{}>
+    pressableState: PressableStateCallbackType
+  }>
+  /**
+   * An optional component to render on the left side of the text.
+   * Example: `LeftAccessory={(props) => <View {...props} />}`
+   */
+  LeftAccessory?: ComponentType<{
+    style: StyleProp<{}>
+    pressableState: PressableStateCallbackType
+  }>
+}
+
+/**
+ * A component that allows users to take actions and make choices.
+ * Wraps the Text component with a Pressable component.
+ *
+ * - [Documentation and Examples](https://github.com/infinitered/ignite/blob/master/docs/Components-Button.md)
+ */
+export function Button(props: ButtonProps) {
+  const {
+    tx,
+    text,
+    style: $viewStyleOverride,
+    pressedStyle: $pressedViewStyleOverride,
+    textStyle: $textStyleOverride,
+    pressedTextStyle: $pressedTextStyleOverride,
+    children,
+    RightAccessory,
+    LeftAccessory,
+    ...rest
+  } = props
+
+  const preset: Presets = !!$viewPresets[props.preset] ? props.preset : "default"
+  function $viewStyle({ pressed }) {
+    return [
+      $viewPresets[preset],
+      $viewStyleOverride,
+      !!pressed && [$pressedViewPresets[preset], $pressedViewStyleOverride],
+    ]
+  }
+  function $textStyle({ pressed }) {
+    return [
+      $textPresets[preset],
+      $textStyleOverride,
+      !!pressed && [$pressedTextPresets[preset], $pressedTextStyleOverride],
+    ]
+  }
+
+  return (
+    <Pressable style={$viewStyle} {...rest}>
```

Comment:
Button is now a Pressable component which supports press states making it more customizable.